openpype/hosts/maya/plugins/publish/extract_import_reference.py

```diff
@@ -0,0 +1,124 @@
+import os
+
+from maya import cmds
+
+import pyblish.api
+
+from openpype.pipeline import publish, legacy_io
+from openpype.settings import get_project_settings
+from openpype.hosts.maya.api import lib
+
+
+def _get_project_setting():
+    project_name = legacy_io.active_project()
+    project_setting = get_project_settings(project_name)
+    maya_enabled = (
+        project_setting["maya"]["publish"]["ImportReference"]["enabled"]
+    )
+    use_published = (
+        project_setting["deadline"]["publish"]["MayaSubmitDeadline"]["use_published"] # noqa
+    )
+    if maya_enabled != use_published:
+        return False
+    else:
+        return use_published
+
+
+class ImportReference(publish.Extractor):
+    """
+
+        Extract the scene with imported reference.
+        The temp scene with imported reference is
+        published for rendering if this extractor is activated
+
+    """
+
+    label = "Import Reference"
+    order = pyblish.api.ExtractorOrder - 0.48
+    hosts = ["maya"]
+    families = ["renderlayer", "workfile"]
+    active = _get_project_setting()
+    optional = True
+    tmp_format = "_tmp"
+
+    def process(self, instance):
+        ext_mapping = (
+            instance.context.data["project_settings"]["maya"]["ext_mapping"]
+        )
+        if ext_mapping:
+            self.log.info("Looking in settings for scene type ...")
+            # use extension mapping for first family found
+            for family in self.families:
+                try:
+                    self.scene_type = ext_mapping[family]
+                    self.log.info(
+                        "Using {} as scene type".format(self.scene_type))
+                    break
+                except KeyError:
+                    # no preset found
+                    pass
```

Comment:
`self.scene_type` might be undefinded if no ext mapping is defined or a keyerror is hit. There currently appears to be no fallback scene type.